twitter_api/main.py

```diff
@@ -1,23 +1,315 @@
+# -*- coding: utf-8 -*-
+import json
 import sqlite3
 
-from flask import Flask
-from flask import g
+from datetime import datetime
+import dateutil.parser as du
+from flask import abort, Flask, g, request, Response
+from hashlib import md5
+from random import choice
+from string import ascii_lowercase, digits
+
+from .utils import auth_only, JSON_MIME_TYPE, json_only
+
 
 app = Flask(__name__)
 
 
+#############################
+# Helper functions:         #
+#############################
 def connect_db(db_name):
     return sqlite3.connect(db_name)
 
-
 @app.before_request
 def before_request():
     g.db = connect_db(app.config['DATABASE'])
 
+def _hash_password(password):
+    '''
+    Returns the MD5 hash of the user's entered password
+    '''
+    # password_bytes = str.encode(password)
+    password_bytes = password.encode('utf-8')
```

Comment:
The encoding depends on the request. A request from Japan might be JIS-encoded instead of UTF-8.